service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/dataproccluster/startup.sh

```diff
@@ -20,6 +20,7 @@
 # Execution details:
 #   By default, this script is executed as root on all Dataproc vm nodes on every startup.
 #   However, the script will exit early if it's not running on the Dataproc manager node and also if it's not the first time the script is run.
+#   NOTE: This script is executed before the dataproc startup script that installs software components and jupyter.
```

Comment:
what is the dataproc startup script?

Reply:
The script itself isn't documented by gcloud but it comes with the prebuild dataproc gce vm images and does all the google related setup, including installing and launching jupyter if we configure it to (which we do).